src/components/grouped-character/grounded-character.test.tsx

```diff
@@ -0,0 +1,604 @@
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.clear(input);
+  await user.type(input, "1222");
+  await input.setSelectionRange(6, 6);
+  await user.tab();
+
+  expect(input.selectionStart).toBe(7);
+  expect(input.selectionEnd).toBe(7);
+});
+
+test("pressing backspace after a separator moves the cursor backwards to positions", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const MockComponent = () => {
+    const [stateValue, setStateValue] = useState<string>("1231231");
+    return (
+      <GroupedCharacter
+        separator="-"
+        value={stateValue}
+        groups={[2, 2, 4]}
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={(ev) => {
+          setStateValue(ev.target.value.formattedValue);
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.keyboard("{arrowLeft}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(4);
+  expect(input.selectionEnd).toBe(4);
+
+  await user.keyboard("{backspace}");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(2);
+  expect(input.selectionEnd).toBe(2);
+});
+
+test("adding a character before a separator moves cursor forwards two positions", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const MockComponent = () => {
+    const [stateValue, setStateValue] = useState<string>("123123");
+    return (
+      <GroupedCharacter
+        separator="-"
+        value={stateValue}
+        groups={[2, 2, 4]}
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={(ev) => {
+          setStateValue(ev.target.value.formattedValue);
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.keyboard("{arrowLeft}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(5);
+  expect(input.selectionEnd).toBe(5);
+
+  await user.keyboard("1");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(7);
+  expect(input.selectionEnd).toBe(7);
+});
+
+test("when no character is present, cursor position goes to 0", async () => {
```

Comment:
**comment** (non-blocking): I'm not sure this test is really adding much value. Perhaps it's for coverage, in which case I guess it's fine

Reply:
Yes it is for coverage. I think your comment has raised a valid point that I should probably put a comment of sorts here saying it's for coverage as it does seem pointless to have.